sources/academy/webscraping/web_scraping_for_beginners/crawling/scraping_the_data.md

```diff
@@ -11,221 +11,327 @@ slug: /web-scraping-for-beginners/crawling/scraping-the-data
+## Extracting data from a product detail page {#extracting-data}
 
-Here's a quick reminder of earlier's data extraction code:
+The term product detail page (or PDP) is commonly used on e-commerce websites to describe the page where you can find detailed information about a product. In the Warehouse store, it's, for example, [this page describing Denon headphones](https://warehouse-theme-metal.myshopify.com/products/denon-ah-c720-in-ear-headphones).
+
+Let's start writing a script that extracts data from this single PDP. We can use this familiar code as boilerplate.
 
 ```js
-// main.js
+// product.js
 import { gotScraping } from 'got-scraping';
 import cheerio from 'cheerio';
 
-const response = await gotScraping('https://demo-webstore.apify.org/search/on-sale');
+const productUrl = 'https://warehouse-theme-metal.myshopify.com/products/denon-ah-c720-in-ear-headphones';
+const response = await gotScraping(productUrl);
 const html = response.body;
 
 const $ = cheerio.load(html);
-const products = $('a[href*="/product/"]');
 
-const results = [];
+// Attribute extraction code will go here.
+```
+
+We will use the techniques learned in the [Basics of data extraction](../data_extraction/index.md) section to find and extract the following product attributes:
+
+- title
+- vendor
+- price
+- number of reviews
+- description
+
+![scraping-title.png](./images/scraping-title.png)
+
+> For brevity, we won't explain how to extract every attribute step-by-step. Review the [Basics of data extraction](../data_extraction/index.md) section to learn about DevTools and extracting data.
+
+### Title
+
+Getting the title is quite straightforward. We recommend using `h1` for titles where available, because it's the semantically correct way and therefore unlikely to change.
+
+```js
+const title = $('h1').text().trim();
+```
+
+### Vendor
+
+Vendor name is available as a link with the `product-meta__vendor` class. We're only interested in the text though.
+
+```js
+const vendor = $('a.product-meta__vendor').text().trim();
+```
+
+### Price
+
+We will take a shortcut here and only extract the price as a string that includes currency. In production scrapers, you might want to split it into two fields.
+
+```js
+const price = $('span.price').contents()[2].nodeValue;
+```
+
+### Number of reviews
+
+For the review count, we use the `parseInt()` function to get only the number. Otherwise, we would receive a string like **2 reviews** from this element.
+
+```js
+const reviewCount = parseInt($('span.rating__caption').text());
+```
+
+### Description
+
+Getting the description is fairly straightforward as well, but notice the two selectors separated by a space: `div[class*="description"]` `div.rte`. This is called a [descendant combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator), and it allows you to search for child elements within parent elements. Using any of the selectors separately would lead to unwanted strings in our result.
```

Comment:
isnt it better to keep this as a single codeblock? 
```suggestion
Getting the description is fairly straightforward as well, but notice the two selectors separated by a space: `div[class*="description"] div.rte`. This is called a [descendant combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator), and it allows you to search for child elements within parent elements. Using any of the selectors separately would lead to unwanted strings in our result.
```